File: LeetCode_1262/Main.py
from typing import List

class Solution:
    def maxSumDivThree(self, nums: List[int]) -> int:
        tot = sum(nums)
        b, c = [], []
        
        for x in nums:
            if x % 3 == 1:
                b.append(x)
            elif x % 3 == 2:
                c.append(x)
        
        b.sort()
        c.sort()

        if tot % 3 == 0:
            return tot

        if tot % 3 == 1:
            op1 = tot - b[0] if len(b) > 0 else 0
            op2 = tot - c[0] - c[1] if len(c) > 1 else 0
            return max(op1, op2)
        else:
            op1 = tot - b[0] - b[1] if len(b) > 1 else 0
            op2 = tot - c[0] if len(c) > 0 else 0
            return max(op1, op2)

File: LeetCode_1262/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxSumDivThree(vector<int>& nums) {
        int tot = 0;
        vector<int> b, c;
        for (int x : nums) {
            tot += x;
            if (x % 3 == 1) b.push_back(x);
            else if (x % 3 == 2) c.push_back(x);
        }
        
        sort(b.begin(), b.end());
        sort(c.begin(), c.end());
        
        if (tot % 3 == 0) return tot;
        
        int ans = 0;
        if (tot % 3 == 1) {
            int op1 = b.size() > 0 ? tot - b[0] : 0;
            int op2 = c.size() > 1 ? tot - c[0] - c[1] : 0;
            ans = max(op1, op2);
        } else {
            int op1 = b.size() > 1 ? tot - b[0] - b[1] : 0;
            int op2 = c.size() > 0 ? tot - c[0] : 0;
            ans = max(op1, op2);
        }
        
        return ans;
    }
};
